### DISCUSSION TITLE
Need Guidance for Retrigger (Falling Edge) GP14 by using RP 2040 PICO

### DISCUSSION THREAD

**Discussion by hasamsiddiqui** [earlephilhower/arduino-pico#2318] [General]
I use 4 PICOs RP2040, a total of 16 ADC channels (4 ADC channels
[Collector2a (1).txt](https://github.com/user-attachments/files/16486955/Collector2a.1.txt)
![updated ADC circuit board Layout (2)](https://github.com/user-attachments/assets/85952f19-e40f-49e0-a1fc-caebb14ab2c5)
 per PICO) where the trigger signal (reference voltage) is connected to GP14. I’ve programmed the Picomite to set a trigger value for a falling edge, which means the trigger should activate when the signal reaches 0.8 V. However, I’m encountering an issue where the output voltages I’m getting are higher than expected—they should be lower or close to 0.8 V. The trigger signal frequency is set at 360 Hz, and the input voltage is a 1.2 V AC sine wave at the same frequency. Currently, the output voltages from all ADC channels are between 0.85 and 0.87 V. When I increase the frequency to 400 Hz, 450 Hz, or 500 Hz, the output voltages rise even higher, which is confusing. Could you please help me identify where I might be making a mistake?
I aim to capture the input signal precisely at the point where it crosses the trigger voltage of 0.8 V on a falling edge. To achieve this, I set the offset to zero (minimum). I connected an oscilloscope in DC coupling mode on the input side to monitor the amplitude and frequency of the input voltage. Since the trigger is set for the falling edge, which means the system should capture the signal as it transitions from rising to falling and reaches the trigger level, I expect the captured voltage to be less than 0.8 V.
Kind regards

Hasham

**Comment by earlephilhower** [comment]
You probably would get a better response at the [Raspberry Pi forums](https://forums.raspberrypi.com/) than here since this is a general HW interface question.

Off my head I'm left wondering why you're using 4 Picos with low resolution (9.5 bits IIRC from the datasheet errata)  and slow ADCs (plus, it is physically impossible to sample more than 1 pin at a time so your "trigger on 0-cross" can never actually sample them simultaneously).  I'm sure there are tons of ADCs with a trigger input you could use instead, and then grab their results using a single Pico.  Or 16 sample-and-hold op amps thru a 16:1 mux and a single Pico or real ADC, for example.
